Makefile

```diff
@@ -138,25 +134,28 @@ import:
 	test -n "$(NAME)" # $$NAME
 	psql $(NAME) < db/schema.sql
 
-
 # Docker actions
-## Rinkeby docker environment
-RINKEBY_COMPOSE_FILE=dockerfiles/rinkeby/docker-compose.yml
-
-.PHONY: rinkeby_env_up
-rinkeby_env_up:
-	docker-compose -f $(RINKEBY_COMPOSE_FILE) up -d geth
-	docker-compose -f $(RINKEBY_COMPOSE_FILE) up --build migrations
-	docker-compose -f $(RINKEBY_COMPOSE_FILE) up -d --build vulcanizedb
-
-.PHONY: rinkeby_env_deploy
-rinkeby_env_deploy:
-	docker-compose -f $(RINKEBY_COMPOSE_FILE) up -d --build vulcanizedb
-
-.PHONY: dev_env_migrate
-rinkeby_env_migrate:
-	docker-compose -f $(RINKEBY_COMPOSE_FILE) up --build migrations
-
-.PHONY: rinkeby_env_down
-rinkeby_env_down:
-	docker-compose -f $(RINKEBY_COMPOSE_FILE) down
+# Build any docker image in dockerfiles
+.PHONY: dockerbuild
+dockerbuild:
+	test -n "$(IMAGE)" # $$IMAGE
+	docker build -t $(IMAGE) -f dockerfiles/$(IMAGE)/Dockerfile .
+
+.PHONY: headersync
+headersync: STARTING_BLOCK_NUMBER ?= 10000000
+headersync: HOST ?= host.docker.internal
+headersync: DATABASE_PASSWORD ?= postgres
+headersync:
+	test -n "$(NAME)" # $$(NAME) - Database Name
+	test -n "$(CLIENT_IPCPATH)" # $$(CLIENT_IPCPATH)
+	docker run \
+		-it \
+		-p "5432:5432" \
+		-e "STARTING_BLOCK_NUMBER=$(STARTING_BLOCK_NUMBER)" \
+		-e "DATABASE_NAME=$(NAME)" \
+		-e "DATABASE_HOSTNAME=$(HOST)" \
+		-e "DATABASE_PORT=$(PORT)" \
+		-e "DATABASE_USER=$(USER)" \
+		-e "DATABASE_PASSWORD=$(DATABASE_PASSWORD)" \
+		-e "CLIENT_IPCPATH=$(CLIENT_IPCPATH)" \
+		header_sync:latest
```

Comment:
Probably a nitpick that's not super relevant to this PR but - do you think it might be confusing that this command is running the latest docker image for header sync? Thinking you might expect to be running the local binary (esp if you're making changes in dev)

Reply:
I think you're right. If you look closely header_sync:latest isn't the makerdao latest (doesn't have the full url), it's the latest that you built if you use the dockerbuild command above.

What I think I should do here is have the image be a command line param, so that's explicit.

Reply:
I should add - I'm double checking this to ensure it's not pulling down from maker, as that wasn't the point of this task.

Reply:
Double-checked. After removing my many..many ...many header_sync:latest containers (ask me about my awk script) this task won't run, because headersync isn't local. Still gonna allow setting the image name though.

```
make headersync NAME=vulcanize_development CLIENT_IPCPATH=https://geth0.mainnet.makerfoundation.com:8545 
test -n "vulcanize_development" # $(NAME) - Database Name
test -n "https://geth0.mainnet.makerfoundation.com:8545" # $(CLIENT_IPCPATH)
docker run \
		-it \
		-p "5432:5432" \
		-e "STARTING_BLOCK_NUMBER=10000000" \
		-e "DATABASE_NAME=vulcanize_development" \
		-e "DATABASE_HOSTNAME=host.docker.internal" \
		-e "DATABASE_PORT=5432" \
		-e "DATABASE_USER=postgres" \
		-e "DATABASE_PASSWORD=postgres" \
		-e "CLIENT_IPCPATH=https://geth0.mainnet.makerfoundation.com:8545" \
		header_sync:latest
Unable to find image 'header_sync:latest' locally
docker: Error response from daemon: pull access denied for header_sync, repository does not exist or may require 'docker login': denied: requested access to the resource is denied.
See 'docker run --help'.
make: *** [headersync] Error 125
```

Reply:
Added a configurable name for the make task, but also verified that header_sync:latest doesn't somehow get the version on dockerhub.